src/Confluent.Kafka.Avro/AvroDeserializer.cs

```diff
@@ -36,21 +37,30 @@ namespace Confluent.Kafka.Serialization
     /// </remarks>
     public class AvroDeserializer<T>
     {
-        private ISchemaRegistryClient schemaRegistryClient;
-
         private IAvroDeserializerImpl<T> deserializerImpl;
 
 
         /// <summary>
         ///     Initiliaze a new AvroDeserializer instance.
         /// </summary>
-        /// <param name="schemaRegisteryClient">
-        ///     An instance of an implementation of ISchemaRegistryClient used for
-        ///     communication with Confluent Schema Registry.
+        /// <param name="config">
+        ///     Deserializer configuration properties.
         /// </param>
-        public AvroDeserializer(ISchemaRegistryClient schemaRegisteryClient)
+        public AvroDeserializer(IEnumerable<KeyValuePair<string, string>> config = null)
         {
-            schemaRegistryClient = schemaRegisteryClient;
+            if (config == null) { return; }
+
+            var nonAvroConfig = config.Where(item => !item.Key.StartsWith("avro."));
+            if (nonAvroConfig.Count() > 0)
+            {
+                throw new ArgumentException($"AvroDeserializer: unexpected configuration parameter {nonAvroConfig.First().Key}");
+            }
+
+            var avroConfig = config.Where(item => item.Key.StartsWith("avro."));
+            if (avroConfig.Count() != 0)
+            {
+                throw new ArgumentException($"AvroDeserializer: unexpected configuration parameter {avroConfig.First().Key}");
```

Comment:
i'm on the fence here. not a bad idea but susceptible to the out-of-date comment problem, but worse since it's in the code. I think i'd prefer to leave as is.